package/libgluonutil/src/libgluonutil.c

```diff
@@ -109,6 +109,34 @@ struct json_object * gluonutil_wrap_and_free_string(char *str) {
 	return ret;
 }
 
+
+bool gluonutil_get_mesh_prefix(struct in6_addr *result) {
+	struct json_object *site = gluonutil_load_site_config();
+	if (!site)
+		return false;
+
+	struct json_object *node_prefix = NULL;
+	if (!json_object_object_get_ex(site, "node_prefix6", &node_prefix))
+		return false;
+
+	const char *str_prefix = json_object_get_string(node_prefix);
+	if (!str_prefix)
+		return false;
+
+	char *prefix_addr = strndup(str_prefix, strchrnul(str_prefix, '/')-str_prefix);
+
+	int rc = inet_pton(AF_INET6, prefix_addr, result);
+
+	free(prefix_addr);
+	json_object_put(site);
+
+	if (rc == 1 )
+		return true;
+
+	return false;
```

Comment:
Just a minor stylistic commend: I think it would be nice to have all the `return false` in if-branches, so the unconditional return at the end of the function is the `return true`.

Reply:
it does make sense to keep this the same style. done.